core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -95,6 +97,33 @@ export class RubricsEditorComponent {
     return Boolean(this.editableExplanations[difficulty][index]);
   }
 
+  isExplanationLengthValid(difficulty: string, idx: number): boolean {
+    return this.editableExplanations[difficulty][idx].length <=
+    this.maximumCharacterLengthOfExplanation;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    // Checking if medium level rubrics have atleast one explantion.
+    this.selectedRubricIndex = 1;
```

Comment:
Why are we attaching this to the "this" scope? It could just be a local variable.

Reply:
Perhaps you could use a const variable "MEDIUM_EXPLANATION_INDEX" at the top of this component and use it whereever you need to access the medium explanation.

Reply:
Yeah, we can do that, the reason why I did this is that in this file wherever we are accessing the rubric index we are accessing this variable to set the value and then use it, we can change it though.
Thanks

Reply:
Oh but isn't this method just supposed to check if the medium explanation is valid? If so, it shouldn't be setting any global variable (seems like a "side effect").

Reply:
Yes, you are right, thanks for the clarification. I'll change that.